cockatrice/src/pixmapgenerator.cpp

```diff
@@ -112,26 +112,35 @@ QPixmap CountryPixmapGenerator::generatePixmap(int height, const QString &countr
 
 QMap<QString, QPixmap> CountryPixmapGenerator::pmCache;
 
-QPixmap UserLevelPixmapGenerator::generatePixmap(int height, UserLevelFlags userLevel, bool isBuddy)
+QPixmap UserLevelPixmapGenerator::generatePixmap(int height, UserLevelFlags userLevel, bool isBuddy, QString privLevel)
 {
-
-    int key = height * 10000 + (int) userLevel + (int) isBuddy;
+    int privkeysize = privLevel.size();
+    if (privLevel.size() == 0)
+        privkeysize = 1;
+    
+    int key = height * 10000 + (int)userLevel + (int)isBuddy + privkeysize;
```

Comment:
This can easily collide as userLevel, isBuddy, and privkeysize are all single numbers... Try changing to something like `int key = height * 10000 + (int)userLevel*88 + (int)isBuddy*54 + privkeysize*31;`